script/bulk_import/generic_bulk.rb

```diff
@@ -464,6 +458,13 @@ def import_user_emails
       user_id = user_id_from_imported_id(row["id"])
       next if user_id && existing_user_ids.include?(user_id)
 
+      if row["anonymized"] == 1
+        username = username_from_id(user_id)
+        email_prefix = /\Aanon_?\d+\Z/.match?(username) ? username : "anon_#{anon_username_suffix}"
```

Comment:
Can you explain why this is necessary? Is this because the import script might reuse an existing user based on `external_id` (SSO)?

Reply:
For a user being anonymized, since `import_users` is called before `import_user_emails`, an anonymized username should be available via the username_from_id method during the `import_user_emails` step.  In this case we directly  reuse the already anonymized username for the email prefix.

A new `email_prefix`  only gets generated  if for some reason(possibly due to  incremental imports)  a user  initially gets  imported as normal and subsequent runs have this user marked for anonymization. In this case, we still want to anonymize the email hence the check and call to `anon_username_suffix`

Reply:
That's a weird edge case and it would leave the user in a not fully anonymized state. Also, I'm pretty sure that an incremental import wouldn't even import emails because of `next if user_id && existing_user_ids.include?(user_id)`

Reply:
Yes I agree. I know it's an extreme case. Do you think it's pretty safe to go without the extra check and assume the `username`  is already anonymized by the time `import_user_emails ` get's called?

Reply:
Yes, we can drop the extra check. Our current implementation doesn't modify existing users, so the check doesn't do anything.